### PR TITLE
Cleanup binary masks typing

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Add typing for binary masks
* Make naming more evident

Was confused about our image/masks handling and cleaned it up a bit.

## How has it been tested?

* CI
* Manual test

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_commands/train_task_helpers.py [hunk 1]
```diff
@@ -202,20 +207,23 @@ def get_dataset(
...
+        # During validation every image can have a different shape.
+        "image": torch.stack(images) if split == "train" else images,
+        "mask": torch.stack(masks) if split == "train" else masks,
+        "binary_masks": [item["binary_masks"] for item in batch],
     }
 
-    # During training images and masks all have the same shape.
-    if split == "train":
-        out["image"] = torch.stack(out["image"])
-        out["mask"] = torch.stack(out["mask"])
```
Comment (guarin) [inline src/lightly_train/_commands/train_task_helpers.py:217]: Move it further up for better typing

File: src/lightly_train/_data/mask_semantic_segmentation_dataset.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ class MaskSemanticSegmentationDataArgs(TaskDataArgs):
     train: SplitArgs
     val: SplitArgs
     classes: dict[int, str]
-    ignore_classes: set[int] | None = None
+    ignore_classes: set[int] | None = Field(default=None, strict=False)
```
Comment (guarin) [inline src/lightly_train/_data/mask_semantic_segmentation_dataset.py:232]: Not sure how this got lost but this fixesingore classes.

File: src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py [hunk 3]
```diff
@@ -395,47 +395,6 @@ def validation_step(
             },
         )
 
-    def get_targets(self, masks: Tensor) -> list[dict[str, Tensor]]:
-        # This follows logic from: https://github.com/tue-mps/eomt/blob/716cbd562366b9746804579b48b866da487d9485/datasets/ade20k_semantic.py#L47-L48
-        targets = []
-        for mask in masks:
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py:423]: Unused

File: src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py [hunk 4]
```diff
@@ -395,47 +395,6 @@ def validation_step(
...
-                {
-                    "masks": torch.stack(img_masks),
-                    "labels": mask.new_tensor(img_labels, dtype=torch.long),
-                }
-            )
-        return targets
-
-    @torch.compiler.disable  # type: ignore[misc]
-    def to_per_pixel_targets_semantic(
-        self,
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py:441]: Unused

Comment (guarin) [inline src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/train_model.py:419]: Unused

File: src/lightly_train/types.py [hunk 5]
```diff
@@ -48,18 +48,18 @@ class TransformOutputSingleView(TypedDict):
...
 # The type and variable names of the Batch is fully determined by the type and
 # variable names of the DatasetItem by the dataloader collate function.
 class Batch(TypedDict):
     filename: list[ImageFilename]  # length==batch_size
-    views: list[Tensor]  # One tensor per view, of shape (batch_size, 3, w, h) each.
+    views: list[Tensor]  # One tensor per view, of shape (batch_size, 3, H, W) each.
     masks: NotRequired[
         list[Tensor]
-    ]  # One tensor per view, of shape (batch_size, w, h) each.
+    ]  # One tensor per view, of shape (batch_size, H, W) each.
```
Comment (guarin) [inline src/lightly_train/types.py:62]: Comments were wrong, not sure why. The conversion from (H, W, C) to (C, H, W) happens in the transform.
